### ISSUE TITLE
[css-paint-api] Allow custom paint to optionally declare an intrinsic size.

### ISSUE THREAD

**Issue by bfgeek** [w3c/css-houdini-drafts#31]
This would allow for things like:
 - border-image.
 - tiling of background image.
 - keeping aspect ratio.

Should this be declared ahead of time, or set within a callback?

**Comment by frivoal** [comment]
Ahead of time seems simpler. Do we have any other animated image type whose intrinsic size/ratio can change on each frame? If yes, maybe within the callback is fine, but that seems to introduce unnecessary complications for little to no benefit.

**Comment by tabatkins** [comment]
Yes, we do have such an image type - `crossfade()`, when given two images with different intrinsic dimensions, transitions those dimensions as its own.

So we do indeed need some way to declare that your dimensions can change. On the other hand, the 99% case is dimensions *not* changing.

How about:

* Add a "dimensions" key to PaintInitializionOptions, which accepts a `(PaintDimensionsDict or DOMString)`.  PaintDimensionsDict should be a dict with width/height/aspectratio keys.  The accepted DOMString is "dynamic".  Omitting implies no dimensions.
* Add a "sizeCallback" key to PaintInitializationOptions.  It takes the same params as paintCallback, and is expected to return a PaintDimensionsDict.  We call it any time we call paintCallback (before or after doesn't matter, it's just important that they always be called together).  It is only called if you passed "dynamic" to the "dimensions" key.

**Comment by frivoal** [comment]
@tabatkins, that sounds reasonable.

**Comment by bfgeek** [comment]
Moving this to lvl-2 at the moment as per f2f. We can revisit once we've had some implementation experience. Specified that they have no intrinsic dimensions in 2b7a93bc9f867455c0653bd7edf1164049389ca1 .
